### PR TITLE
Updating libs/scripts to require Java 11

### PR DESCRIPTION
**Pull request by bendk**
I saw an email from Grisha indicating Java 11 is needed to build fenix.  That's what I saw too when trying to build yesterday, so I updated some scripts and documentation.

Looks like the taskcluster config also needs to be updated, but I'm not sure how to do that.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: libs/verify-android-ci-environment.sh [hunk 1]
```diff
@@ -53,8 +53,8 @@ location of your Java installation."
 fi
 
 JAVA_VERSION=$("$JAVACMD" -version 2>&1 | grep -i version | cut -d'"' -f2 | cut -d'.' -f1-2)
-if [[ "${JAVA_VERSION}" != "1.8" ]]; then
-  echo "Incompatible java version: ${JAVA_VERSION}. JDK 8 must be installed."
+if [[ "${JAVA_VERSION}" != "1.11" ]]; then
+  echo "Incompatible java version: ${JAVA_VERSION}. JDK 11 must be installed."
   echo "Try switching versions and re-running. Using sdkman: sdk install java 8.0.282+8.hs-adpt || sdk use java 8.0.282+8.hs-adpt"
```
Comment (bendk) [inline libs/verify-android-ci-environment.sh:58]: This line should be updated as well, but I don't use `sdkman` and I'm a bit afraid of installing it and having 1 more thing that installs JDKs to me system.  Does anyone else have a suggestion for this?

**Comment by rfk** [comment]
> Looks like the taskcluster config also needs to be updated, but I'm not sure how to do that.

I believe the logic for controlling the taskcluster build config lives in [`./taskcluster/docker/linux/Dockerfile`](https://github.com/mozilla/application-services/blob/main/taskcluster/docker/linux/Dockerfile) but I do not have any experience with updating it. If we're lucky, changes to that file will automagically be picked up by taskcluster!

**Comment by bendk** [comment]
So I tried updating the java version in the `apt-get` command there and ran into this issue: https://firefox-ci-tc.services.mozilla.com/tasks/TeELKfxlSKai_w2UetuVrg/runs/0/logs/public/logs/live.log

I'm guessing that code that runs `sdkmanager` needs to be updated, but it's not so obvious how to do it.  This combined with #4369 makes me want to rewrite that code.

**Comment by pocmo** [comment]
> I'm guessing that code that runs `sdkmanager` needs to be updated, but it's not so obvious how to do it. This combined with #4369 makes me want to rewrite that Dockerfile.

We ran into this in the A-C repository too. Our workaround for this is to install jdk 8 and 11, and for sdkmanager calls explicitly use Java 8:
- https://github.com/mozilla-mobile/android-components/blob/main/taskcluster/docker/base/Dockerfile#L35-L36
- https://github.com/mozilla-mobile/android-components/blob/6c0fe91ab83614265f533b819bf406fc046b2227/taskcluster/scripts/decision-install-sdk.sh#L19

File: libs/verify-android-ci-environment.sh (hunk 1 above)
Comment (grigoryk) [reply libs/verify-android-ci-environment.sh:58]: `sdk install java 11.0.3.hs-adpt || sdk use 11.0.3.hs-adpt`

Comment (grigoryk) [reply libs/verify-android-ci-environment.sh:58]: sdk install java 11.0.3.hs-adpt || sdk use 11.0.3.hs-adpt

**Comment by bendk** [comment]
Got a "looks reasonable" from Agi, can someone from a-s approve this PR?

**Pull request merged**